core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtilsSuite.scala

```diff
@@ -35,6 +36,41 @@ class MesosSchedulerUtilsSuite extends SparkFunSuite with Matchers with MockitoS
     val sc = mock[SparkContext]
     when(sc.conf).thenReturn(sparkConf)
   }
+
+  def createTestPortResource(range: (Long, Long), role: Option[String] = None): Resource = {
```

Comment:
Yes we could but usually test code shouldnt reuse code under testing...anyway...